### PR TITLE
Feature vet tec #779

### LINKED ISSUE #779: Add VET-TEC to code schools page
# Feature request
Add VET-TEC as a category under Schools.

## Proposed solution
Currently, each code school card has an entry for the GI bill. Solution involves;
* An entry for both the GI bill and VET-TEC.
* A filter at the top of the page allowing filtering for VET-TEC as a category.

## Additional context
`isVetTecApproved` has been added as a boolean-typed field for the each code school in the code_schools endpoint.

https://api.operationcode.org/docs/

### PR DESCRIPTION
**Pull request by mgrandori90**
Resolves #779 

# Description of changes
This PR really only represents one thing: the ability to use the pre-existing "Schools Accepting Vet Tec" button to filter those schools. With that, there are a couple other small changes:

- A separate card banner for Vet Tec approved schools on the SchoolCard

- A small excerpt about Vet Tec (from VA.org) on the code schools page

- Knobs for storybook's SchoolCard for Vet Tec

# Issue Resolved

-  An entry for both the GI bill and VET-TEC.

- A filter at the top of the page allowing filtering for VET-TEC as a category.

Fixes #{779}

# Other todos / considerations

- Just as one of the members had flipped some of the boolean values for me last night, all of the correct schools need to be flipped (I believe there is 8 of them with multiple locations). Right now I believe it is just three random schools out of the AllSchools list.

-I was trying to keep it simple but I believe there are other design options and ideas for the banners. I also believe that having a dropdown filter for the categories might be beneficial once there are more and more programs available.

## Screenshots/GIFs

Screen Shot 2019-09-23 at 8 19 15 PM
Screen Shot 2019-09-23 at 8 19 27 PM
Screen Shot 2019-09-23 at 8 19 34 PM
Screen Shot 2019-09-23 at 8 19 59 PM
Screen Shot 2019-09-23 at 8 26 13 PM
Screen Shot 2019-09-23 at 8 38 46 PM

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 0c7fcbaec0ff3fc8b2031365b3e0ae6977222d74

https://deploy-preview-796--operation-code-storybook.netlify.com

File: components/Cards/SchoolCard/SchoolCard.css [hunk 1]
```diff
@@ -65,6 +65,80 @@
   left: 100%;
 }
 
+.vetTecRibbon {
```
Comment (kylemh) [inline components/Cards/SchoolCard/SchoolCard.css]: The only good critique I can really add is toward reducing the amount of CSS being written.

It seems like there exists enough here to create a `.ribbon` class and then you can the exclusive subset of rules like:

```css
.ribbon .vetTech {}
.ribbon .giBill {}
.ribbon .dual {}
```

File: pages/code_schools.js [hunk 2]
```diff
@@ -104,14 +104,13 @@ export default class CodeSchools extends React.Component {
...
-
     this.setState({ filteredSchools: vaApproved, selectedStates: [] });
   };
 
   filterVetTecApproved = () => {
     const { allSchools } = this.props;
-    const vetTecApprovedSchools = allSchools.filter(school => school.isVetTecApproved);
-    this.setState({ filteredSchools: vetTecApprovedSchools, selectedStates: [] });
+    const vetTecApproved = allSchools.filter(school => school.isVetTecApproved);
+    this.setState({ filteredSchools: vetTecApproved, selectedStates: [] });
```
Comment (kylemh) [inline pages/code_schools.js]: I'd prefer the old name because the data structure is embedded as part of it (schools - an array of schools), but you don't need to revert this change. Just explaining why it was done that way!

File: components/Cards/SchoolCard/SchoolCard.css (hunk 1 above)
Comment (mgrandori90) [reply components/Cards/SchoolCard/SchoolCard.css]: Thank you for this idea. It certainly simplified the ribbons specific classes and will make it easy to add others or change them in the future.

File: pages/code_schools.js (hunk 2 above)
Comment (mgrandori90) [reply pages/code_schools.js]: I see why you chose the old name, and I do think the old name is more fitting. When I looked at it originally I was trying to make it resemble the vaApproved filter as much as I could since they both represent filters for a specific program- but now I know the data for vaApproved is dependent on location, the data for vetTec is dependent on the school. Makes sense! Thanks. Reverted back to the OG!

**Comment by kylemh** [comment]
Wow! Thanks so much for your contribution. It means the world to us here at Operation Code.

As a token of our gratitude, please fill out [this form](https://goo.gl/forms/r02wt0pBNhkxYciI3) to receive some stickers and a handwritten "thank you" note. ❤️
